src/main/java/io/inversion/cloud/model/Relationship.java

```diff
@@ -73,6 +73,42 @@ public Entity getRelated()
       return related;
    }
 
+   public Relationship getInverse()
+   {
+      if (!isManyToMany())
+      {
+         for (Relationship other : related.getRelationships())
+         {
+            if (isOneToMany() && !other.isManyToOne())
+               continue;
+
+            if (isManyToMany() && !other.isOneToMany())
+               continue;
+
+            if (getFkIndex1().equals(other.getFkIndex1()) //
+                  && getPrimaryKeyTable1().getPrimaryIndex().equals(other.getPrimaryKeyTable1().getPrimaryIndex()))
+            {
+               return other;
+            }
+         }
+      }
+      else if (isManyToMany())
```

Comment:
is this check necessary?  since the if above is !isManyToMany()?

Reply:
you are right...that was an artifact of readability when I was working through the first iteration.  I've reversed the order of the if/else and removed the second if to make it more readable.